From 8fc846e3179fbe83085988b31adf8ca17cf72add Mon Sep 17 00:00:00 2001
From: Sharaf Zaman <sharafzaz121@gmail.com>
Date: Wed, 28 Oct 2020 08:53:23 +0000
Subject: [PATCH 41/43] Android: don't use getExternalStorageDirectory and
 getExternalStoragePublicDirectory

These methods have been deprecated for apps targetting Sdk 29+
---
 src/corelib/io/qstandardpaths_android.cpp | 27 ++++++-----------------
 1 file changed, 7 insertions(+), 20 deletions(-)

diff --git a/src/corelib/io/qstandardpaths_android.cpp b/src/corelib/io/qstandardpaths_android.cpp
index 9fe98d34cd..0ac558f8c2 100644
--- a/src/corelib/io/qstandardpaths_android.cpp
+++ b/src/corelib/io/qstandardpaths_android.cpp
@@ -188,18 +188,11 @@ QString QStandardPaths::writableLocation(StandardLocation type)
 {
     switch (type) {
     case QStandardPaths::MusicLocation:
-        return getExternalFilesDir("DIRECTORY_MUSIC");
     case QStandardPaths::MoviesLocation:
-        return getExternalFilesDir("DIRECTORY_MOVIES");
     case QStandardPaths::PicturesLocation:
-        return getExternalFilesDir("DIRECTORY_PICTURES");
     case QStandardPaths::DocumentsLocation:
-        if (QtAndroidPrivate::androidSdkVersion() > 18)
-            return getExternalFilesDir("DIRECTORY_DOCUMENTS");
-        else
-            return getExternalFilesDir() + QLatin1String("/Documents");
     case QStandardPaths::DownloadLocation:
-        return getExternalFilesDir("DIRECTORY_DOWNLOADS");
+        return standardLocations(type).first();
     case QStandardPaths::GenericConfigLocation:
     case QStandardPaths::ConfigLocation:
     case QStandardPaths::AppConfigLocation:
@@ -229,36 +222,30 @@ QString QStandardPaths::writableLocation(StandardLocation type)
 QStringList QStandardPaths::standardLocations(StandardLocation type)
 {
     if (type == MusicLocation) {
-        return QStringList() << writableLocation(type)
-                             << getExternalFilesDir("DIRECTORY_MUSIC")
+        return QStringList() << getExternalFilesDir("DIRECTORY_MUSIC")
                              << getExternalFilesDir("DIRECTORY_PODCASTS")
                              << getExternalFilesDir("DIRECTORY_NOTIFICATIONS")
                              << getExternalFilesDir("DIRECTORY_ALARMS");
     }
 
     if (type == MoviesLocation) {
-        return QStringList() << writableLocation(type)
-                             << getExternalFilesDir("DIRECTORY_MOVIES");
+        return QStringList() << getExternalFilesDir("DIRECTORY_MOVIES");
     }
 
     if (type == PicturesLocation) {
-        return QStringList()  << writableLocation(type)
-                              << getExternalFilesDir("DIRECTORY_PICTURES");
+        return QStringList() << getExternalFilesDir("DIRECTORY_PICTURES");
     }
 
     if (type == DocumentsLocation) {
         if (QtAndroidPrivate::androidSdkVersion() > 18) {
-            return QStringList() << writableLocation(type)
-                                 << getExternalFilesDir("DIRECTORY_DOCUMENTS");
+            return QStringList() << getExternalFilesDir("DIRECTORY_DOCUMENTS");
         } else {
-            return QStringList() << writableLocation(type)
-                                 << getExternalFilesDir() + QLatin1String("/Documents");
+            return QStringList() << getExternalFilesDir() + QLatin1String("/Documents");
         }
     }
 
     if (type == DownloadLocation) {
-        return QStringList() << writableLocation(type)
-                             << getExternalFilesDir("DIRECTORY_DOWNLOADS");
+        return QStringList() << getExternalFilesDir("DIRECTORY_DOWNLOADS");
     }
 
     if (type == AppDataLocation || type == AppLocalDataLocation) {
-- 
2.20.1.windows.1

